assets/src/hooks/useFetchTimepoints.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from "react"
+import { fetchTimepointsForRoute } from "../api"
+import { RouteId, Timepoint, TimepointsByRouteId } from "../skate"
+import {
+  Dispatch,
+  setLoadingTimepointsForRoute,
+  setTimepointsForRoute,
+} from "../state"
+
+export const useFetchTimepoints = (
+  selectedRouteIds: RouteId[],
+  timepointsByRouteId: TimepointsByRouteId,
+  dispatch: Dispatch
+): void => {
+  useEffect(() => {
+    selectedRouteIds.forEach((routeId: RouteId) => {
+      if (!(routeId in timepointsByRouteId)) {
+        dispatch(setLoadingTimepointsForRoute(routeId))
+
+        fetchTimepointsForRoute(routeId).then((newTimepoints: Timepoint[]) =>
+          dispatch(setTimepointsForRoute(routeId, newTimepoints))
+        )
+      }
+    })
+  }, [selectedRouteIds])
```

Comment:
this seems like it also depends on `timepointsByRouteId`?

Reply:
It does, but we don't care about rerunning the effect if `timeointsByRouteId` changes. And if we do include it as a dependency, the effect will rerun (without doing anything) whenever we receive new data.

Given that, should `timepointsByRouteId` still be in the list of dependencies?

Reply:
My instinct is that `timepointsByRouteId` shouldn't be a dependency. It's true that it's used here, but in effect it is the result of this function, so we know that if it has changed it is because we just changed it. It is something we should keep in our heads because it will be a concern if `timepointsByRouteId` is ever altered by something else in the future, but I can't imagine a scenario for that happening right now.